topic-operator/src/main/java/io/strimzi/operator/topic/v2/TopicOperatorMain.java

```diff
@@ -0,0 +1,171 @@
+            throw new IllegalStateException();
+        }
+
+        shutdownHook = new Thread(this::shutdown, "TopicOperator-shutdown-hook");
+        LOGGER.infoOp("Installing shutdown hook");
+        Runtime.getRuntime().addShutdownHook(shutdownHook);
+        LOGGER.infoOp("Starting health and metrics");
+        healthAndMetricsServer.start();
+        LOGGER.infoOp("Starting queue");
+        queue.start();
+        informer = Crds.topicOperation(client)
+                .inNamespace(namespace)
+                // Do NOT use withLabels to filter the informer, since the controller is stateful
+                // (topics need to be added to removed from TopicController.topics if KafkaTopics transition between
+                // selected and unselected).
+                .runnableInformer(resyncIntervalMs)
+                .addEventHandlerWithResyncPeriod(handler, resyncIntervalMs)
+                .itemStore(itemStore);
+        LOGGER.infoOp("Starting informer");
+        informer.run();
+    }
+
+    synchronized void stop() {
+        if (shutdownHook == null) {
+            throw new IllegalStateException();
+        }
+        // shutdown(), will be be invoked indirectly by calling
+        // h.run() has the side effect of nullifying this.shutdownHook
+        // so retain a reference now so we have something to call
+        // removeShutdownHook() with.
+        var h = shutdownHook;
+        // Call run (not start()) on the thread so that shutdown() is executed
+        // on this thread.
+        shutdown();
+        // stop() is _not_ called from the shutdown hook, so calling
+        // removeShutdownHook() should not cause IAE.
+        Runtime.getRuntime().removeShutdownHook(h);
+    }
+
+    private synchronized void shutdown() {
+        // Note: This method can be invoked on either via the shutdown hook thread or
+        // on the thread(s) on which stop()/start() are called
+        LOGGER.infoOp("Shutdown initiated");
+        try {
+            shutdownHook = null;
+            if (informer != null) {
+                informer.stop();
+                informer = null;
+            }
+            this.queue.stop();
+            this.healthAndMetricsServer.stop();
+            LOGGER.infoOp("Shutdown completed normally");
+        } catch (InterruptedException e) {
+            LOGGER.infoOp("Interrupted during shutdown");
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Entrypoint for unidirectional TO.
+     * @param args Command line args
+     * @throws Exception If bad things happen
+     */
+    public static void main(String[] args) throws Exception {
+        TopicOperatorConfig topicOperatorConfig = TopicOperatorConfig.buildFromMap(System.getenv());
+        TopicOperatorMain operator = operator(topicOperatorConfig, kubeClient(), Admin.create(topicOperatorConfig.adminClientConfig()));
+        operator.start();
+        LOGGER.infoOp("Returning from main()");
```

Comment:
Just to be clear: `operator.start()` returns, rather than blocks, so it's normal to see this log message when the operator starts, and seeing it tells you that startup is complete. The TO process doesn't terminate when the main thread stops because by that point the non-daemon queue processing threads are running. 

I think if we make this `DEBUG` we should probably make many/all of the other log messages in this class `DEBUG` too. But TBH they're all only printed once anyway, so I think info is fine, personally.